databuilder/extractor/restapi/github_file_extractor.py

```diff
@@ -45,30 +47,32 @@ def init(self, conf):
         self.file_urls = None
 
     def extract(self):
-        # type: () -> Any
-
+        # type: () -> Union[None, Tuple[str, str, str]]
         """
-        Fetch one result row from RestApiQuery, convert to {model_class} if specified before
-        returning.
-        :return:
+        :return: (filename, filepath, file contents as a string)
         """
         if not self._extract_iter:
             self._extract_iter = self._get_extract_iter()
 
         try:
-            file_url = next(self._get_extract_iter())
-            file_contents = self._client.get_file_contents_from_url(file_url)
-            yield file_contents
+            (name, path, download_url) = next(self._get_extract_iter())
+            file_contents = self._client.get_file_contents_from_url(download_url)
+            return (name, path, file_contents)
```

Comment:
moving this comment to a place where it makes sense: changed `yield` to `return` because it was inconsistent with other extractors, although I'd agree that `yield` would be better here.